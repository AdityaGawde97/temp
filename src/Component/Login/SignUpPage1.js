import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { styles } from '../../Css/LoginCss'
import { OutlinedTextField } from 'react-native-material-textfield'
import { Button } from 'react-native-elements';

export default class SignUpPage1 extends Component {
    constructor(props) {
        super(props);
        this.state = {
            firstName: '',
            lastName: '',
            errors: {}
        };
    }

    validateForm = () => {
        let errors = {};
        let formIsValid = true;

        if(!this.state.firstName) {
            errors["firstName"] = "*Enter first name";
            formIsValid = false;
        }

        if(!this.state.lastName) {
            errors["lastName"] = "*Enter last name";
            formIsValid = false;
        }

        this.setState({
            errors: errors
        })

        return formIsValid
    }

    goToSignUpPage2 = (event) => {
        event.preventDefault();
        if(this.validateForm()){
            this.props.navigation.navigate('SignUpPage2',{
                firstName: this.state.firstName,
                lastName: this.state.lastName
            })
        }
    }
    
    render() {

        return (
            <View style={styles.mainContainer}>

                <Text style={styles.appTitle}>
                    <Text style={{ color: '#4285F4' }}>F</Text>
                    <Text style={{ color: '#DB4437' }}>u</Text>
                    <Text style={{ color: '#F4B400' }}>n</Text>
                    <Text style={{ color: '#4285F4' }}>d</Text>
                    <Text style={{ color: '#0F9D58' }}>o</Text>
                    <Text style={{ color: '#DB4437' }}>o</Text>
                </Text>

                <Text style={styles.createTitle}>
                    Create a Fundoo Account
                </Text>

                <Text style={styles.subTitle}>
                    Enter your name
                </Text>

                <View style={styles.inputContainer}>


                    <View style={styles.inputView}>
                        <OutlinedTextField
                            label="First name"
                            value={this.state.firstName}
                            onChangeText={(firstName)=>this.setState({firstName})}
                            error={this.state.errors.firstName}
                            errorColor='red'
                        />
                    </View>

                    <View style={styles.inputView2}>
                        <OutlinedTextField
                            label="Last name"
                            value={this.state.lastName}
                            onChangeText={(lastName)=>this.setState({lastName})}
                            error={this.state.errors.lastName}
                            errorColor='red'
                        />
                    </View> 

                </View>

                <View style={styles.nextInSignUp}>
                        <Button 
                            title="Next"
                            onPress={this.goToSignUpPage2}
                        />
                </View>

            </View>
        );
    }
}
