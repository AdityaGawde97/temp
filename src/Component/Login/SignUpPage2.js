import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { styles } from '../../Css/LoginCss'
import { OutlinedTextField } from 'react-native-material-textfield'
import { Button } from 'react-native-elements';

export default class SignUpPage2 extends Component {
    constructor(props) {
        super(props);
        
        this.state = {
            email: '',
            errors: {}
        }

    }

    validateForm = () => {
        let errors = {};
        let formIsValid = true;
        if (typeof this.state.email !== "undefined") {
            let pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
            if (!pattern.test(this.state.email)) {
                formIsValid = false;
                errors["email"] = "*Enter valid email-ID.";
            }
        }

        if (!this.state.email) {
            errors["email"] = "*Enter email id";
            formIsValid = false;
        }

        this.setState({
            errors: errors
        })

        return formIsValid
    }

    goToSignUpPage3 = (event) => {
        event.preventDefault();
        if (this.validateForm()) {
            const { navigation } = this.props;
            let firstName = navigation.getParam('firstName');
            let lastName = navigation.getParam('lastName');
            this.props.navigation.navigate('SignUpPage3', {
                firstName: firstName,
                lastName: lastName,
                email: this.state.email
            })
        }
    }

    render() {



        return (
            <View style={styles.mainContainer}>

                <Text style={styles.appTitle}>
                    <Text style={{ color: '#4285F4' }}>F</Text>
                    <Text style={{ color: '#DB4437' }}>u</Text>
                    <Text style={{ color: '#F4B400' }}>n</Text>
                    <Text style={{ color: '#4285F4' }}>d</Text>
                    <Text style={{ color: '#0F9D58' }}>o</Text>
                    <Text style={{ color: '#DB4437' }}>o</Text>
                </Text>

                <Text style={styles.createTitle}>
                    Choose your Fundoo address
                </Text>

                <Text style={styles.subTitle}>
                    Create your own
                </Text>

                <View style={styles.inputContainer}>

                    <View style={styles.inputView}>
                        <OutlinedTextField
                            label="Email"
                            value={this.state.email}
                            onChangeText={(email) => this.setState({ email })}
                            error={this.state.errors.email}
                            errorColor='red'
                        />
                    </View>

                </View>

                <View style={styles.nextInSignUp}>
                    <Button
                        title="Next"
                        onPress={this.goToSignUpPage3}
                    />
                </View>

            </View>
        );
    }
}