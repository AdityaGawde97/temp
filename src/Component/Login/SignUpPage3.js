import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { styles } from '../../Css/LoginCss'
import { OutlinedTextField } from 'react-native-material-textfield'
import { Button } from 'react-native-elements';

export default class SignUpPage2 extends Component {
    constructor(props) {
        super(props);

        this.state = {
            password: '',
            confirm: '',
            visibleIcon: false,
            errors: {}
        };
    }

    validateForm = () => {
        let errors = {};
        let formIsValid = true;

        if (typeof this.state.password !== "undefined") {
            if (!this.state.password.match(/^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).*$/)) {
                formIsValid = false;
                errors["password"] = "*Enter secure and strong password.";
            }
        }

        if (!this.state.password) {
            errors["password"] = "*Enter password";
            formIsValid = false;
        }

        if (typeof this.state.confirm !== "undefined") {
            if (this.state.confirm !== this.state.password) {
                formIsValid = false;
                errors["confirm"] = "*Password does not match.";
            }
        }

        if (!this.state.confirm) {
            errors["confirm"] = "*Enter password";
            formIsValid = false;
        }


        this.setState({
            errors: errors
        })

        return formIsValid
    }

    submitUserSignUp = (event) => {
        event.preventDefault();
        if (this.validateForm()) {

        }
    }

    render() {

        return (
            <View style={styles.mainContainer}>

                <Text style={styles.appTitle}>
                    <Text style={{ color: '#4285F4' }}>F</Text>
                    <Text style={{ color: '#DB4437' }}>u</Text>
                    <Text style={{ color: '#F4B400' }}>n</Text>
                    <Text style={{ color: '#4285F4' }}>d</Text>
                    <Text style={{ color: '#0F9D58' }}>o</Text>
                    <Text style={{ color: '#DB4437' }}>o</Text>
                </Text>

                <Text style={styles.createTitle}>
                    Create a strong password
                </Text>

                <Text
                    style={{
                        flex: 0,
                        textAlign: 'center',
                        fontSize: 16,
                        marginLeft: 35,
                        marginRight: 35,
                        top: 60
                    }}>
                    Create a strong password with a mix of letters, numbers and symbols
                </Text>



                <View style={styles.inputContainer}>

                    <View style={styles.passIconView}>
                        <View style={styles.passView}>
                            <OutlinedTextField
                                label="Password"
                                value={this.state.password}
                                onChangeText={(password) => this.setState({ password })}
                                secureTextEntry={!this.state.visibleIcon}
                                error={this.state.errors.password}
                                errorColor='red'
                            />
                        </View>
                        <View style={{ padding: 12 }}>
                            <TouchableOpacity onPress={ () => this.setState({visibleIcon:!this.state.visibleIcon})}>
                                {
                                    !this.state.visibleIcon ?
                                        <Image
                                            style={styles.passIcon}
                                            source={require('../../Assets/visible.png')}
                                        /> :
                                        <Image
                                            style={styles.passIcon}
                                            source={require('../../Assets/off.png')}
                                        />
                                }

                            </TouchableOpacity>
                        </View>
                    </View>

                    <View style={styles.passView2}>
                        <OutlinedTextField
                            label="Confirm"
                            value={this.state.confirm}
                            onChangeText={(confirm) => this.setState({ confirm })}
                            secureTextEntry={!this.state.visibleIcon}
                            error={this.state.errors.confirm}
                            errorColor='red'
                        />
                    </View>



                </View>

                <View style={styles.nextInSignUp}>
                    <Button
                        title="Next"
                        onPress={this.submitUserSignUp}
                    />
                </View>

            </View>
        );
    }
}