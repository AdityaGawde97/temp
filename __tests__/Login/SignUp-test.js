import React from 'react';
import { shallow } from 'enzyme';
import SignUpPage1 from "../../src/Component/Login/SignUpPage1";
import SignUpPage2 from "../../src/Component/Login/SignUpPage2";
import SignUpPage3 from "../../src/Component/Login/SignUpPage3";

import renderer from 'react-test-renderer';

/**************************** SignUpPage1 Component Testing ******************************/

// This test just uses Jest snapshot testing
it('SignUpPage1 Page renders correctly, test using Jest', () => {
    renderer.create(<SignUpPage1 />);
});

// Using Jest + Enzyme
describe('<SignUpPage1/>', () => {
    it('SignUp Page renders correctly, test using Jest + Enzyme', () => {
        expect(shallow(<SignUpPage1 />)).toMatchSnapshot();
    });

    const wrapper = shallow(<SignUpPage1 />)

    it('checking initial state', () => {
        expect(wrapper.state('firstName')).toEqual('');
        expect(wrapper.state('lastName')).toEqual('');
    })
});
