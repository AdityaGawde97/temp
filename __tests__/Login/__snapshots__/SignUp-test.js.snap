// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage1/> SignUp Page renders correctly, test using Jest + Enzyme 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontFamily": "Roboto",
        "fontSize": 26,
        "top": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      F
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      u
    </Text>
    <Text
      style={
        Object {
          "color": "#F4B400",
        }
      }
    >
      n
    </Text>
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      d
    </Text>
    <Text
      style={
        Object {
          "color": "#0F9D58",
        }
      }
    >
      o
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      o
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 27,
        "top": 50,
      }
    }
  >
    Create a Fundoo Account
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 20,
        "top": 60,
      }
    }
  >
    Enter your name
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 9,
        "flexDirection": "column",
        "top": 100,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0,
          "width": 370,
        }
      }
    >
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="First name"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "top": 15,
          "width": 370,
        }
      }
    >
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="Last name"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "flex": 0,
        "padding": 35,
        "width": 180,
      }
    }
  >
    <ForwardRef(Themed.Button)
      onPress={[Function]}
      title="Next"
    />
  </View>
</View>
`;
